src/extensions/agent_modules/cloud_communication/eg_mqtt_broker_client/src/upd_request_operation.c

```diff
@@ -481,13 +484,14 @@ static void HandleReportResults(ADUC_Retriable_Operation_Context* context, ADUC_
                 free(updData->reportingJson);
                 updData->reportingJson = NULL;
                 ADUC_StateStore_SetReportResultsAck(false);
-                AduUpdUtils_TransitionState(ADU_UPD_STATE_REPORT_RESULTS_ACK, updData);
+                AduUpdUtils_TransitionState(ADU_UPD_STATE_REPORT_RESULTS_ACK, updData, context);
             }
         }
     }
 }
 
-static void HandleReportResultsAck(ADUC_Retriable_Operation_Context* context, ADUC_Update_Request_Operation_Data* updData, time_t nowTime)
+static void HandleReportResultsAck(
+    ADUC_Retriable_Operation_Context* context, ADUC_Update_Request_Operation_Data* updData, time_t nowTime)
```

Comment:
input validation on updData just in case unless expecting to fail loudly (Please just resolve these if we're expecting a loud failure in the void functions. Just trying to cover the bases)